spv/createenvelope_test.go

```diff
@@ -1133,7 +808,13 @@ func TestSPVEnvelope_CreateEnvelope(t *testing.T) {
 			if test.expErr == nil {
 				assert.NoError(t, err)
 				assert.NotNil(t, envelope)
-				assert.Equal(t, test.exp, *envelope)
+
+				f, err := os.Open("../testing/data/spv/create/" + test.expFile + ".json")
```

Comment:
Use fs.Embed instead - saves having to read files etc and can also be used to auto gen these too I think

Reply:
Good shout

Reply:
Undid this as it breaks compatibility with versions <1.16, which isn't worth in for cleaner test code IMO

Reply:
I think we should use the same compatibility as the go team for libs and only support that last 2 versions of go, in this case 1.17 and 1.16 [pub](https://golang.org/doc/devel/release#policy)

This is why I tend to only have test for last 2 versions

Reply:
Our build system supports the passed 3 tho

Reply:
Chatted offline, 1.15 is going the way of the dodo